File: tests/src-android/cgeo/geocaching/settings/TestSettings.java
package cgeo.geocaching.settings;

import cgeo.geocaching.R;


/**
 * provide write-access proxy to settings for testing purposes
 */
public final class TestSettings extends Settings {

    /**
     * Purely static!
     */
    private TestSettings() {
        throw new InstantiationError();
    }

    public static void setLogin(final Credentials credentials) {
        Settings.setLogin(credentials.getUsernameRaw(), credentials.getPasswordRaw());
    }

    public static void setUseImperialUnits(final boolean imperial) {
        putBoolean(R.string.pref_units_imperial, imperial);
    }

    public static void setCacheTwitterMessage(final String template) {
        putString(R.string.pref_twitter_cache_message, template);
    }

    public static void setTrackableTwitterMessage(final String template) {
        putString(R.string.pref_twitter_trackable_message, template);
    }

    public static void setSignature(final String signature) {
        putString(R.string.pref_signature, signature);
    }

}

File: main-kotlin/src/androidTest/kotlin/cgeo/geocaching/settings/TestSettings.kt
// Auto-converted from Java to Kotlin
// WARNING: This code requires manual review and likely has compilation errors
// Please review and fix:
// - Method signatures (parameter types, return types)
// - Field declarations without initialization
// - Static members (use companion object)
// - Try-catch-finally blocks
// - Generics syntax
// - Constructors
// - And more...

package cgeo.geocaching.settings

import cgeo.geocaching.R


/**
 * provide write-access proxy to settings for testing purposes
 */
class TestSettings : Settings() {

    /**
     * Purely static!
     */
    private TestSettings() {
        throw InstantiationError()
    }

    public static Unit setLogin(final Credentials credentials) {
        Settings.setLogin(credentials.getUsernameRaw(), credentials.getPasswordRaw())
    }

    public static Unit setUseImperialUnits(final Boolean imperial) {
        putBoolean(R.string.pref_units_imperial, imperial)
    }

    public static Unit setSignature(final String signature) {
        putString(R.string.pref_signature, signature)
    }

}
